myia/ir/node.py

```diff
@@ -74,6 +74,28 @@ def apply(self, fn, *inputs):
 
         return Apply(self, *edges)
 
+    def clone(self, objmap=None):
+        """Make a copy of this graph."""
+        res = Graph(self.parent)
+        if objmap is None:
+            objmap = {self: res}
+        else:
+            objmap[self] = res
```

Comment:
We should probably check if the graph is already in the objmap, if e.g. we use it more than once.

```suggestion
        if objmap is None:
            objmap = {self: res}
        elif self in objmap:
            return objmap[self]
        else:
            objmap[self] = res
```